src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/InterceptedProjectProperties.cs

```diff
@@ -37,6 +37,30 @@ public InterceptedProjectProperties(ImmutableArray<Lazy<IInterceptingPropertyVal
             _valueProviders = builder.ToImmutable();
         }
 
+        public override async Task<bool> IsValueInheritedAsync(string propertyName)
+        {
+            if (!_valueProviders.TryGetValue(propertyName, out Lazy<IInterceptingPropertyValueProvider, IInterceptingPropertyValueProviderMetadata>? valueProvider))
+            {
+                return await base.IsValueInheritedAsync(propertyName);
+            }
+
+            if (valueProvider.Value is IInterceptingPropertyValueProvider2 valueProviderValueWithMsBuildProperties)
+            {
+                string[]? msBuildPropertyNames = (await valueProviderValueWithMsBuildProperties.GetProjectFileMSBuildPropertyNamesAsync(propertyName, DelegatedProperties))?.ToArray();
+                if (msBuildPropertyNames?.Length > 0)
+                {
+                    string[] propertiesDefinedInProjectFile = (await GetDirectPropertyNamesAsync()).ToArray();
+                    if (msBuildPropertyNames.Any(name => propertiesDefinedInProjectFile.Contains(name)))
```

Comment:
Why this instead of calling `base.IsValueInheritedAsync(...)` for each of the `msBuildPropertyNames`?

Reply:
that can end up doing a lot of reads